homeassistant/components/unifi/device_tracker.py

```diff
@@ -193,6 +196,18 @@ def add_entities(controller, async_add_entities, tracked):
         async_add_entities(new_tracked)
 
 
+@callback
+def remove_entities(controller, mac_addresses, tracked, entity_registry):
+    """Remove select tracked entities."""
+    for mac in mac_addresses:
+
+        if mac not in tracked:
+            continue
+
+        entity = tracked.pop(mac)
+        controller.hass.async_create_task(entity.async_remove())
```

Comment:
I think we can make this simpler by signaling each entity to remove itself instead of signaling these callbacks outside the entities. The entity knows its client mac address and knows what features the entity supports, wired or not etc.

If we would change this generally, we could also avoid storing the entity in `tracked` and instead just store the mac addresses in a set.

Generally we prefer to use the entity api from inside the entity.

Reply:
I think it leads to double book keeping, right now its a single source of truth and control

Reply:
It's still a single source of truth. The signaller decides what should happen. The entity takes care of what it is its responsibility, nothing more.

Reply:
You mean the tracked set should be passed into all entities?

Reply:
That works, but if we want to avoid making the entities responsible directly for `tracked` we can pass them a callback that they should call when they remove themselves. This callback would handle modifying `tracked`.

I imagine we create signals using the mac addresses. The entities register to listen for their own mac address. The entity has access to the controller and the client and the entity knows its type. So the check if it's time to remove an entity can be moved to each entity class. The entity can store a cached controller option as an instance attribute and compare that with the updated controller option when receiving the signal.

Reply:
Ok, I will give it a shot